lib/plugins/aws/package/compile/events/httpApi/index.js

```diff
@@ -351,17 +358,32 @@ Object.defineProperties(
           }
           const routeConfig = { targetData: routeTargetData };
           if (authorizer) {
-            const { name, scopes } = (() => {
+            const { name, scopes, id } = (() => {
               if (_.isObject(authorizer)) return authorizer;
               return { name: authorizer };
             })();
-            if (!authorizers.has(name)) {
+            if (id) {
+              if (!userConfig.id) {
+                throw new this.serverless.classes.Error(
+                  `Event references external authorizer '${id}', but httpApi is part of the current stack.`,
+                  'EXTERNAL_HTTP_API_AUTHORIZER_WITHOUT_EXTERNAL_HTTP_API'
+                );
+              }
+              if (name) {
+                throw new this.serverless.classes.Error(
+                  `Event references external authorizer and provide a local name property in function ${functionName}.`,
```

Comment:
Let's skip on that validation (this would be in scope of https://github.com/serverless/serverless/issues/6562 and general idea is to keep core logic free from validation that can easily be covered by JSON schema based validation)